arroyo/processing/processor.py

```diff
@@ -102,15 +105,25 @@ def on_partitions_revoked(partitions: Sequence[Partition]) -> None:
             [topic], on_assign=on_partitions_assigned, on_revoke=on_partitions_revoked
         )
 
-    def __commit(self, positions: Mapping[Partition, Position]) -> None:
+    def __commit(
+        self,
+        positions: Mapping[Partition, Position],
+        throttle_sec: Optional[float] = None,
+    ) -> None:
         self.__consumer.stage_positions(positions)
-        start = time.time()
-        self.__consumer.commit_positions()
-        logger.debug(
-            "Waited %0.4f seconds for offsets to be committed to %r.",
-            time.time() - start,
-            self.__consumer,
-        )
+
+        now = time.time()
+
+        if throttle_sec is not None and (
+            self.__last_committed is None or now - self.__last_committed > throttle_sec
+        ):
```

Comment:
I think if `throttle_sec` is None this function will never commit. The logic here should be something like:

```
if throttle_sec exists and the time threshold has not been hit:
    return

self.__consumer.commit_positions()
...

Reply:
You're right, thanks